packages/block-editor/src/components/block-tools/block-selection-button.js

```diff
@@ -193,6 +193,13 @@ function BlockSelectionButton( { clientId, rootClientId } ) {
 			selectBlock( startingBlockClientId );
 			setBlockMovingClientId( null );
 		}
+		if (
+			startingBlockClientId &&
+			selectedBlockClientId === startingBlockClientId &&
+			navigateIn
+		) {
+			return;
+		}
```

Comment:
Let's add the comment above this check so it's clear why we're returning early.

Reply:
Done ✅